### PR TITLE
Provide additional method for Base64::encode.

### PR DESCRIPTION
**Pull request by RomanDzhabarov**

### CODE REVIEW

File: source/common/common/base64.h [hunk 1]
```diff
@@ -12,11 +12,28 @@ class Base64 {
...
    * Note, decoded string may contain '\0' at any position, it should be treated as a sequence of
    * bytes.
    */
   static std::string decode(const std::string& input);
+
+private:
+  /**
+   * Helper methods for encoding.
+   */
+  static inline void encode_base(const uint8_t cur_char, uint64_t pos, uint8_t& next_c,
```
Comment (mattklein123) [inline source/common/common/base64.h]: del inline here and below

File: source/common/common/base64.h [hunk 2]
```diff
@@ -12,11 +12,28 @@ class Base64 {
   static std::string encode(const Buffer::Instance& buffer, uint64_t length);
 
   /**
+   * Base64 encode an input string.
+   * @param input string to encode.
+   */
+  static std::string encode(const std::string& input);
```
Comment (mattklein123) [inline source/common/common/base64.h]: make this take a const char* and a length, so it's usable outside of a string.

File: source/common/common/base64.h (hunk 1 above)
Comment (mattklein123) [inline source/common/common/base64.h]: encodeBase and encodeLast

File: source/common/common/base64.h [hunk 3]
```diff
@@ -12,11 +12,28 @@ class Base64 {
...
    * @param input supplies the input to decode.
    *
    * Note, decoded string may contain '\0' at any position, it should be treated as a sequence of
    * bytes.
    */
   static std::string decode(const std::string& input);
+
+private:
+  /**
+   * Helper methods for encoding.
```
Comment (mattklein123) [inline source/common/common/base64.h]: nit: method

File: source/common/common/base64.h [hunk 4]
```diff
@@ -12,11 +12,28 @@ class Base64 {
...
+   */
+  static void encodeBase(const uint8_t cur_char, uint64_t pos, uint8_t& next_c, std::string& ret);
+  /**
```
Comment (mattklein123) [inline source/common/common/base64.h]: nit: newline before this line

File: source/common/common/base64.h [hunk 5]
```diff
@@ -12,11 +12,28 @@ class Base64 {
...
    * Note, decoded string may contain '\0' at any position, it should be treated as a sequence of
    * bytes.
    */
   static std::string decode(const std::string& input);
+
+private:
+  /**
+   * Helper methods for encoding.
+   */
+  static void encodeBase(const uint8_t cur_char, uint64_t pos, uint8_t& next_c, std::string& ret);
```
Comment (rshriram) [inline source/common/common/base64.h]: nit: What does this method do? (you have comment for encodeLast, but none for encodeBase)

Comment (RomanDzhabarov) [inline source/common/common/base64.h]: fixed

File: source/common/common/base64.h (hunk 4 above)
Comment (RomanDzhabarov) [inline source/common/common/base64.h]: fixed

File: source/common/common/base64.h (hunk 3 above)
Comment (RomanDzhabarov) [inline source/common/common/base64.h]: fixed

**Pull request merged**